src/rdf4cpp/Timezone.hpp

```diff
@@ -130,23 +132,144 @@ struct Timezone {
 
 using OptionalTimezone = std::optional<Timezone>;
 
-using TimePoint = std::chrono::time_point<std::chrono::local_t, std::chrono::milliseconds>;
+template<typename I = int64_t>
+struct Year {
+    // adapted from https://howardhinnant.github.io/date_algorithms.html
+    I year = 0;
+
+    [[nodiscard]] constexpr bool is_leap() const noexcept(noexcept(year % 100)) {
+        return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+    }
+
+    constexpr auto operator<=>(Year const &) const noexcept = default;
+};
+
+template<typename Y = int64_t>
+struct Date {
```

Comment:
This corresponds to `chrono::year_month_day` right? If so I think it makes sense to name it `YearMonthDay` for consistency

Reply:
for consistency with std and the `YearMonth` type I would still prefer `YearMonthDay`